src/task_impl/atomic_task.rs

```diff
@@ -52,30 +70,35 @@ impl AtomicTask {
         }
     }
 
-    /// The caller must ensure mutual exclusion
-    pub unsafe fn park(&self) {
-        if let Some(ref task) = *self.task.get() {
```

Comment:
It was never measured. The main point was managing the logic around atomically swapping the consumer task while the producer is notifying. The original API was `unsafe` mostly because it was for a specific use case (`FuturesUnordered`) where we knew there was only a single consumer and it had to be `&self`.

Either way, if it proves to be a perf issue, we can always bring it back as `park_unchecked`.